lib/os/heap.c

```diff
@@ -261,7 +261,6 @@ void *sys_heap_aligned_alloc(struct sys_heap *heap, size_t align, size_t bytes)
 	 * We over-allocate to account for alignment and then free
 	 * the extra allocations afterwards.
 	 */
-	size_t alloc_sz = bytes_to_chunksz(h, bytes);
 	size_t padded_sz = bytes_to_chunksz(h, bytes + align - 1);
```

Comment:
I might be misunderstanding how this works. If I want to allocate 13 bytes aligned on a 16-byte border, I call this function as `sys_heap_aligned_alloc(heap, 16, 13)`, right? So, now `padded_sz = bytes_to_chunksz(h, 29);` If it's a "small heap," we get `(29 + 4 + 7) / 8 = 5`, right? That means we want to allocate 40 bytes for 13 bytes of user data plus 4 bytes of header, if the user wants to align on a 16-byte boundary? Wouldn't 24 bytes be enough for that?